### PR TITLE
docs: use the same variable name for install guide as in deployment

### PR DESCRIPTION
**Pull request by dosmanak**
remove confusing naming of Azure resource group
changes made using:
```
grep --exclude-dir=.git -rl AZURE_NODE_RESOURCE_GROUP|xargs sed -i -e 's/AZURE_NODE_RESOURCE_GROUP/AZURE_RESOURCE_GROUP/'
```

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
synchronize Azure variable name for resource group with cilium operator deployment
```

### CODE REVIEW

**Review by ti-mo — APPROVED** [review]
Thanks, these look like the same thing to me.

File: Documentation/gettingstarted/k8s-install-helm.rst
Comment (nbusseneau) [inline Documentation/gettingstarted/k8s-install-helm.rst:123]: The rationale for not using the same terminology is exactly because of what's stated here: the internal node resource group used for Azure IPAM is not the same as the resource group the user provides when creating the cluster.

When I initially made this change, I thought this was a good way to make it clear they are not the same and improve user understanding of what's going on, and still think the same, so I would be inclined not to change it and rather am curious as to what you find confusing. Could you elaborate? Maybe there is a way to improve this.

**Review by nbusseneau — CHANGES_REQUESTED** [review]
See below.

**Review by joestringer — APPROVED** [review]
There are no docs structure implications, approving for that codeowner.

It looks like the discussion with Nicolas needs to be resolved above before this PR can move forward.

File: Documentation/gettingstarted/k8s-install-helm.rst
Comment (dosmanak) [reply Documentation/gettingstarted/k8s-install-helm.rst:123]: Hi Nicolas and thank you for your attention to my PR.
Although I understand the note on the resource group the final environment variable used in cilium called `AZURE_RESOURCE_GROUP` is filled with Helm value `azure.resourceGroup`
```
$ grep -r -C1 AZURE_RESOURCE_GROUP charts/cilium
charts/cilium/templates/cilium-operator/deployment.yaml-        {{- if .Values.azure.resourceGroup }}
charts/cilium/templates/cilium-operator/deployment.yaml:        - name: AZURE_RESOURCE_GROUP
charts/cilium/templates/cilium-operator/deployment.yaml-          value: {{ .Values.azure.resourceGroup }}
```
If you want to keep emphasization that azure resource group is meant for the Nodes and not the AKS itself i.e. `nodeResourceGroup` field in the json output of `az cli` I suggest to change the underlying deployment env variable as well to keep the names in sync. But that woudln't be just documentation change and may bring more confusion.

I am not fan of Helm so I am trying to patch generated helm using kustomize and that is why I hit this discrepancy.

Comment (nbusseneau) [reply Documentation/gettingstarted/k8s-install-helm.rst:123]: I personally disagree: the Azure IPAM uses `AZURE_RESOURCE_GROUP` because it does not care over which RG it will operate, so the fact the RG passed in to IPAM is an arbitrary top-level RG or the low-level node RG does not matter from its point of view.

On the other hand, when deploying, the administrator should care about which RG they pass to IPAM, because the lower-level RG is more secure (IPAM constrainted to its scope, cannot interfere with APIs outside of the node RG, which might be an issue if the RG containing the AKS cluster also hosts other Azure objects). Yet, this is not a black and white situation because the administrator might consciously decide to use IPAM at an higher level for some reason (other tools interacting with Azure APIs at this level, etc.).

In this sort of situation, I personally tend to err on the side "document properly and leave it to the human to know what they're doing, but still try to make it easy for someone to just copy/paste if they don't care" => hence how the GSG is structured at the moment.

Does this make sense? What is your opinion on this, considering the above?

(Sorry for the delay in answering this.)

Comment (dosmanak) [reply Documentation/gettingstarted/k8s-install-helm.rst:123]: When I check the source code I find following
https://github.com/cilium/cilium/blob/9a0858e2a70f8103496aa35b6b771ba4c5695128/operator/option/config.go#L193
https://github.com/cilium/cilium/blob/9a0858e2a70f8103496aa35b6b771ba4c5695128/operator/option/config.go#L194

and if I get it correctly, it translates into environment variable `AZURE_RESOURCE_GROUP`, hence it makes sense to use such name even in helm scripts.

Thank you for detailed insight into Azure IPAM.

For me it is confusing the variable used in GSG (getting stared guide?) is not used in the cilium operator itself. However I understand the need to emphasize  that the resource group is different from the one were the **kubernetes service** so I must accept your reasoning. @nbusseneau I let you close the PR.

**Pull request closed without merging**
